accounts-db/src/tiered_storage/meta.rs

```diff
@@ -16,10 +16,33 @@ pub struct AccountMetaFlags {
     pub has_rent_epoch: bool,
     /// whether the account is executable
     pub executable: bool,
-    /// the reserved bits.
-    reserved: B30,
+    /// this fewer-than-u64 lamports info stores lamports that can fit
+    /// within its limitation, or a bit indicating the lamport is stored
+    /// separately as an optional field.
+    ///
+    /// Note that the number of bits using in this field must match
+    /// the const LAMPORTS_INFO_BITS.
+    pub lamports_info: B30,
 }
 
+/// The number of bits used in lamports_info field.
+/// Note that this value must match the bits in AccountMetaFlags::lamports_info.
+pub const LAMPORTS_INFO_BITS: u64 = 30;
+/// The max lamports balance that the lamports_info field can handle.
+/// Any lamports beyond this value will be stored separately in optional fields.
+pub const LAMPORTS_INFO_MAX_BALANCE: u64 =
+    ((1u64 << LAMPORTS_INFO_BITS) - 1) - LAMPORTS_INFO_RESERVED_VALUES;
+
+/// The number of special values inside lamports_info.
+/// This const MUST be updated when adding new reserved values.
+pub const LAMPORTS_INFO_RESERVED_VALUES: u64 = 2;
+
+/// A reserved lamports_info value indicating zero-lamports balance.
+pub const LAMPORTS_INFO_IS_ZERO_BALANCE: u32 = 0;
+/// A reserved lamports_info value indicating the lamports balance is stored
+/// in optional fields.
+pub const LAMPORTS_INFO_HAS_OPTIONAL_FIELD: u32 = 1;
```

Comment:
Here's a good place to start the PR review as it includes most of the high-level concepts.

Reply:
Sounds good.  And I will move those constants above the struct definition.

Reply:
Done